drivers/power_domain/power_domain_intel_adsp.c

```diff
@@ -66,15 +66,15 @@ static int pd_intel_adsp_init(const struct device *dev)
 
 #define DT_DRV_COMPAT intel_adsp_power_domain
 
-#define POWER_DOMAIN_DEVICE(id)                                                                    \
-	static struct pg_registers pd_pg_reg##id = {                                               \
-		.wr_address = DT_REG_ADDR(DT_PHANDLE(DT_DRV_INST(id), lps)) + PWRCTL_OFFSET,       \
-		.rd_address = DT_REG_ADDR(DT_PHANDLE(DT_DRV_INST(id), lps)) + PWRSTS_OFFSET,       \
-		.SPA_bit = DT_INST_PROP(id, bit_position),                                         \
-		.CPA_bit = DT_INST_PROP(id, bit_position),                                         \
-	};                                                                                         \
-	PM_DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_pm_action);                                     \
-	DEVICE_DT_INST_DEFINE(id, pd_intel_adsp_init, PM_DEVICE_DT_INST_GET(id),                   \
+#define POWER_DOMAIN_DEVICE(id)
+	static struct pg_registers pd_pg_reg##id = {
+		.wr_address = DT_REG_ADDR(DT_PHANDLE(DT_DRV_INST(id), dfpmcch)) + PWRCTL_OFFSET,
```

Comment:
`wr_address` and `rd_address` point to PWRCTL/PWRSTS register, which are not in dfpmcch (71B90/71B92 respectively).  As you're defining structs for those regions, use them instead of offsets

Reply:
@kfrydryx & @galak please re-review and resolve the conversation if it is o.k for you ;)